components/size_aware_image.jsx

```diff
@@ -309,30 +309,44 @@ export default class SizeAwareImage extends React.PureComponent {
                     </div>
                     <span
                         className={classNames('image-preview-utility-buttons-container', 'image-preview-utility-buttons-container--small-image', {
-                            'image-preview-utility-buttons-container--small-image-no-copy-button': !enablePublicLink || !this.isInternalImage,
+                            'image-preview-utility-buttons-container--small-image-no-copy-button': !enablePublicLink,
                         })}
+                        // since there is a max-width constraint on images, a max-left clause follows.
+                        style={this.state.imageWidth > MIN_IMAGE_SIZE ? (
+                            // cases with no copy link buttons has different offsets than with copy buttons
+                            !enablePublicLink ? {left : `min(${wideImageButtonsOffset + 8}px, calc(100% - 24px)`}
+                            :  {left : `min(${wideImageButtonsOffset}px, calc(100% - 32px)`}
+                         ) : {}}
```

Comment:
please try to avoid nested ternary expressions, especially when you are using object literals directly in props, since it will have no reference to a stored object and results in unnecessary re-renders.

Better declare it above (just below line 296) and use it here. That way you even have more methods to choose from (e.g. a switch/case).

But I would rather add something like this:

```
const modifier = !enablePublicLink ? 8 : 0;
const styles = this.state.imageWidth > MIN_IMAGE_SIZE ? {
    left: `min(${wideImageButtonsOffset + modifier}px, calc(100% - ${32 - modifier}px)`
} : {};
```

Reply:
thank you @michelengelen for the feedback, some are already present in the latest commit, https://github.com/mattermost/mattermost-webapp/pull/9705/files#diff-83c6734e3d6d99f358f1bbc40bd17e5a478d44eed57d06e7e1af20f4c31aaef0R305 but this is neater, the code-block you suggest, I'll reflect this change there.

Reply:
I'm actually surprised that this wasn't picked up be ESLint. It shouldn't allow nested ternaries, but maybe it has difficulty with that happening in JSX